modules/utxo-lib/src/bitgo/UtxoPsbt.ts

```diff
@@ -293,6 +294,83 @@ export class UtxoPsbt<Tx extends UtxoTransaction<bigint> = UtxoTransaction<bigin
     });
   }
 
+  /**
+   * @returns true if the input at inputIndex is a taproot key path.
+   * Checks for presence of minimum required key path input fields and absence of any script path only input fields.
+   */
+  isTaprootKeyPathInput(inputIndex: number): boolean {
+    const input = checkForInput(this.data.inputs, inputIndex);
+    return (
+      !!input.tapInternalKey &&
+      !!input.tapMerkleRoot &&
+      !(
+        input.tapLeafScript?.length ||
+        input.tapScriptSig?.length ||
+        input.tapBip32Derivation?.some((v) => v.leafHashes.length)
+      )
+    );
+  }
+
+  /**
+   * @returns true if the input at inputIndex is a taproot script path.
+   * Checks for presence of minimum required script path input fields and absence of any key path only input fields.
+   */
+  isTaprootScriptPathInput(inputIndex: number): boolean {
+    const input = checkForInput(this.data.inputs, inputIndex);
+    return (
+      !!input.tapLeafScript?.length &&
+      !(
+        this.getProprietaryKeyVals(inputIndex, {
+          identifier: PSBT_PROPRIETARY_IDENTIFIER,
+          subtype: ProprietaryKeySubtype.MUSIG2_PARTICIPANT_PUB_KEYS,
+        }).length ||
+        this.getProprietaryKeyVals(inputIndex, {
+          identifier: PSBT_PROPRIETARY_IDENTIFIER,
+          subtype: ProprietaryKeySubtype.MUSIG2_PUB_NONCE,
+        }).length ||
+        this.getProprietaryKeyVals(inputIndex, {
+          identifier: PSBT_PROPRIETARY_IDENTIFIER,
+          subtype: ProprietaryKeySubtype.MUSIG2_PARTIAL_SIG,
+        }).length
+      )
+    );
+  }
+
+  /**
+   * @returns true if the input at inputIndex is a taproot
+   */
+  isTaprootInput(inputIndex: number): boolean {
+    const input = checkForInput(this.data.inputs, inputIndex);
+    const isP2TR = (script: Buffer): boolean => {
+      try {
+        getTaprootOutputKey(script);
+        return true;
+      } catch (e) {
+        return false;
+      }
+    };
+    return !!(
+      input.tapInternalKey ||
+      input.tapMerkleRoot ||
+      input.tapLeafScript?.length ||
+      input.tapBip32Derivation?.length ||
+      input.tapScriptSig?.length ||
+      this.getProprietaryKeyVals(inputIndex, {
+        identifier: PSBT_PROPRIETARY_IDENTIFIER,
+        subtype: ProprietaryKeySubtype.MUSIG2_PARTICIPANT_PUB_KEYS,
+      }).length ||
+      this.getProprietaryKeyVals(inputIndex, {
+        identifier: PSBT_PROPRIETARY_IDENTIFIER,
+        subtype: ProprietaryKeySubtype.MUSIG2_PUB_NONCE,
+      }).length ||
+      this.getProprietaryKeyVals(inputIndex, {
+        identifier: PSBT_PROPRIETARY_IDENTIFIER,
+        subtype: ProprietaryKeySubtype.MUSIG2_PARTIAL_SIG,
+      }).length ||
+      (input.witnessUtxo && isP2TR(input.witnessUtxo.script))
+    );
```

Comment:
kind of looks like the term `isTaprootKeyPathInput() || isTaprootScriptPathInput()` is in there